app/controllers/users/registrations_controller.rb

```diff
@@ -43,10 +43,13 @@ def disable_account_creation
     end
 
     def track_registration(form)
-      return analytics.track_event('Account Created', form.user) unless form.email_taken?
-
-      existing_user = User.find_by_email(form.email)
-      analytics.track_event('Registration Attempt with existing email', existing_user)
+      if form.email_taken?
+        existing_user = User.find_by_email(form.email)
+        analytics.track_event('Registration Attempt with existing email', existing_user)
+      else
+        analytics.track_event('Account Created', form.user)
+        create_user_event(:account_created, form.user)
```

Comment:
I want to push back on this one.  I'd rather keep all visible events in this table rather than try to synthesize events on demand.  One reason, for example, is that I have a followup change that limits the number of events per user, so the :account-created event should fall of the list after the N event is created.   Keeping all visible events in one place keeps the separation of user visible vs implementation detail clean (in my mind anyway).